### PR TITLE
Add tests take sequence serialized with bad arguments

### PR DESCRIPTION
**Pull request by Lobotuerk**
Signed-off-by: lobotuerk <jtlorente@ekumenlabs.com>

### CODE REVIEW

File: test_rmw_implementation/test/test_subscription.cpp [hunk 1]
```diff
@@ -377,3 +377,31 @@ TEST_F(CLASSNAME(TestSubscriptionUse, RMW_IMPLEMENTATION), count_mismatched_subs
   EXPECT_EQ(RMW_RET_OK, ret) << rmw_get_error_string().str;
   EXPECT_EQ(0u, publisher_count);
 }
+
+TEST_F(CLASSNAME(TestSubscriptionUse, RMW_IMPLEMENTATION), take_serialized_with_bad_args) {
+  rmw_subscription_allocation_t * null_allocation{nullptr}; // still valid allocation
+  rcutils_allocator_t default_allocator = rcutils_get_default_allocator();
+  bool taken;
```
Comment (hidmic) [inline test_rmw_implementation/test/test_subscription.cpp:384]: @Lobotuerk nit: 
```suggestion
  bool taken = false;
```

File: test_rmw_implementation/test/test_subscription.cpp [hunk 2]
```diff
@@ -377,3 +377,31 @@ TEST_F(CLASSNAME(TestSubscriptionUse, RMW_IMPLEMENTATION), count_mismatched_subs
...
+      &serialized_message, 0lu, &default_allocator)) << rmw_get_error_string().str;
+
+  rmw_ret_t ret = rmw_take_serialized_message(nullptr, &serialized_message, &taken, null_allocation);
+  EXPECT_EQ(RMW_RET_INVALID_ARGUMENT, ret) << rmw_get_error_string().str;
+  rmw_reset_error();
+
+  ret = rmw_take_serialized_message(sub, nullptr, &taken, null_allocation);
+  EXPECT_EQ(RMW_RET_INVALID_ARGUMENT, ret) << rmw_get_error_string().str;
+  rmw_reset_error();
+
```
Comment (hidmic) [inline test_rmw_implementation/test/test_subscription.cpp:397]: @Lobotuerk what if `taken` is `nullptr`?

Comment (Lobotuerk) [reply test_rmw_implementation/test/test_subscription.cpp:410]: Updated on d3107ca

File: test_rmw_implementation/test/test_subscription.cpp (hunk 1 above)
Comment (Lobotuerk) [reply test_rmw_implementation/test/test_subscription.cpp]: updated on d3107ca

**Comment by Lobotuerk** [comment]
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/12398/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/7366/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/10109/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/12311/)

File: test_rmw_implementation/test/test_subscription.cpp [hunk 3]
```diff
@@ -377,3 +377,53 @@ TEST_F(CLASSNAME(TestSubscriptionUse, RMW_IMPLEMENTATION), count_mismatched_subs
...
+TEST_F(CLASSNAME(TestSubscriptionUse, RMW_IMPLEMENTATION), take_serialized_with_bad_args) {
+  rmw_subscription_allocation_t * null_allocation{nullptr};  // still valid allocation
+  rcutils_allocator_t default_allocator = rcutils_get_default_allocator();
+  bool taken = false;
+  rmw_serialized_message_t serialized_message = rmw_get_zero_initialized_serialized_message();
+  rmw_serialized_message_t original_message = serialized_message;
+  ASSERT_EQ(
+    RMW_RET_OK, rmw_serialized_message_init(
+      &serialized_message, 1lu, &default_allocator)) << rmw_get_error_string().str;
+  ASSERT_EQ(
```
Comment (Blast545) [inline test_rmw_implementation/test/test_subscription.cpp:390]: If this asserts fails and exits the test early, the rmw_serialized_message_fini is never called for the first msg. You'll have to use a `OSRF_SCOPE_EXIT` macro to make the fini calls in this scenario

Comment (Lobotuerk) [reply test_rmw_implementation/test/test_subscription.cpp:390]: Could you point me to some example of that?

Comment (Lobotuerk) [reply test_rmw_implementation/test/test_subscription.cpp:396]: Updated on b1fca8c

**Review by Blast545 — APPROVED** [review]

**Comment by Lobotuerk** [comment]
CI jobs up to `test_rmw_implementation` after rebase:
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/12439/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/7406/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/10149/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/12352/)

**Comment by Lobotuerk** [comment]
Merging after CI green

**Pull request merged**